### PR TITLE
feat: remove wagmi/query references

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Initially thought of as a bugfix and a cleaner version of https://github.com/shapeshift/web/pull/7182, but really just a cleanup in the end, as the root cause was something entirely different.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

Low

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

- Do a quick smoke test of RFOX and ensure queries are still fired properly, i.e loading states and data looks sane

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

https://jam.dev/c/6686aa13-bc02-4115-9c6f-bc6fc78ad657

### CODE REVIEW

File: src/pages/RFOX/components/Claim/ClaimSelect.tsx [hunk 1]
```diff
@@ -105,12 +105,13 @@ export const ClaimSelect: FC<ClaimSelectProps & ClaimRouteProps> = ({
     isPending: isUnstakingRequestPending,
     isPaused: isUnstakingRequestPaused,
     isError: isUnstakingRequestError,
-    isSuccess: isUnstakingRequestSuccess,
-    isEnabled: isUnstakingRequestEnabled,
+    error,
```
Comment (gomesalexandre) [inline src/pages/RFOX/components/Claim/ClaimSelect.tsx:108]: remove me before opening

File: src/pages/RFOX/components/Claim/ClaimSelect.tsx [hunk 2]
```diff
@@ -105,12 +105,13 @@ export const ClaimSelect: FC<ClaimSelectProps & ClaimRouteProps> = ({
...
     refetch: refetchUnstakingRequest,
     isRefetching: isUnstakingRequestRefetching,
   } = useGetUnstakingRequestQuery({ stakingAssetAccountAddress })
 
+  console.log({ error })
```
Comment (gomesalexandre) [inline src/pages/RFOX/components/Claim/ClaimSelect.tsx:113]: remove me before opening

File: src/pages/RFOX/hooks/useGetUnstakingRequestQuery.ts [hunk 3]
```diff
@@ -103,13 +103,8 @@ export const useGetUnstakingRequestQuery = <SelectData = UnstakingRequest>({
     queryFn: getUnstakingRequestQueryFn,
     select,
     retry: false,
+    throwOnError: true,
```
Comment (gomesalexandre) [inline src/pages/RFOX/hooks/useGetUnstakingRequestQuery.ts:106]: remove me before opening

File: src/AppProviders.tsx [hunk 4]
```diff
@@ -73,8 +73,8 @@ export function AppProviders({ children }: ProvidersProps) {
                       <WalletConnectV2Provider>
                         <KeepKeyProvider>
                           <ErrorBoundary FallbackComponent={ErrorPage} onError={handleError}>
-                            <WagmiConfig config={wagmiConfig}>
-                              <QueryClientProvider>
+                            <QueryClientProvider>
```
Comment (gomesalexandre) [inline src/AppProviders.tsx:76]: - `<QueryClientProvider />` first for wagmi *not* to import its own `useQuery` first and rug things - fine to revert if we want to as this is really just paranoia
- `<WagmiProvider />` over the deprecated `<WagmiConfig />`

File: src/pages/RFOX/components/Unstake/hooks/useRfoxUnstake.tsx [hunk 5]
```diff
@@ -8,7 +8,6 @@ import { RFOX_PROXY_CONTRACT_ADDRESS } from 'contracts/constants'
 import { useMemo } from 'react'
 import type { UseFormReturn } from 'react-hook-form'
 import { encodeFunctionData } from 'viem'
-import type { UseQueryReturnType } from 'wagmi/query'
```
Comment (gomesalexandre) [inline src/pages/RFOX/components/Unstake/hooks/useRfoxUnstake.tsx:11]: Wrong typings from the wrong place

File: src/pages/RFOX/hooks/useEarnedQuery.ts [hunk 6]
```diff
@@ -1,4 +1,5 @@
-import { skipToken } from '@tanstack/react-query'
+import { skipToken, useQuery } from '@tanstack/react-query'
+import type { ReadContractQueryKey } from '@wagmi/core/query'
```
Comment (gomesalexandre) [inline src/pages/RFOX/hooks/useEarnedQuery.ts:2]: `@wagmi/core/query` vs. `wagmi/query` previously as referenced in the [wagmi docs](https://wagmi.sh/core/api/actions/readContract#tanstack-query)

File: src/pages/RFOX/hooks/useGetUnstakingRequestCountQuery.ts [hunk 7]
```diff
@@ -1,4 +1,5 @@
 import { skipToken, useQuery } from '@tanstack/react-query'
+import type { ReadContractQueryKey } from '@wagmi/core/query'
```
Comment (gomesalexandre) [inline src/pages/RFOX/hooks/useGetUnstakingRequestCountQuery.ts:2]: type syntax change as well as using the right place as mentioned in the [docs](https://wagmi.sh/core/api/actions/readContract#tanstack-query).
Effectively ensures type import only (paranoia against evaluating things at runtime and getting into trouble with versions mismatch)

File: src/pages/RFOX/hooks/useStakingBalanceOfQuery.ts [hunk 8]
```diff
@@ -1,12 +1,12 @@
 import { type AssetId, fromAssetId } from '@shapeshiftoss/caip'
-import { skipToken } from '@tanstack/react-query'
+import { skipToken, useQuery } from '@tanstack/react-query'
+import type { ReadContractQueryKey } from '@wagmi/core/query'
```
Comment (gomesalexandre) [inline src/pages/RFOX/hooks/useStakingBalanceOfQuery.ts:3]: type syntax change as well as using the right place as mentioned in the [docs](https://wagmi.sh/core/api/actions/readContract#tanstack-query).
Effectively ensures type import only (paranoia against evaluating things at runtime and getting into trouble with versions mismatch)

File: src/pages/RFOX/hooks/useStakingBalanceOfQuery.ts [hunk 9]
```diff
@@ -68,5 +68,5 @@ export const useStakingBalanceOfQuery = <SelectData = StakingBalanceOf>({
     select,
   })
 
-  return stakingBalanceOfQuery
+  return { ...stakingBalanceOfQuery, queryKey }
```
Comment (gomesalexandre) [inline src/pages/RFOX/hooks/useStakingBalanceOfQuery.ts:71]: Not using `wagmi/query`'s `useQuery` variant anymore, so we expose queryKey the same like `wagmi/query` does

File: src/pages/RFOX/hooks/useStakingInfoQuery.ts [hunk 10]
```diff
@@ -63,5 +63,5 @@ export const useStakingInfoQuery = <SelectData = StakingInfo>({
     select,
   })
 
-  return stakingInfoQuery
+  return { ...stakingInfoQuery, queryKey }
```
Comment (gomesalexandre) [inline src/pages/RFOX/hooks/useStakingInfoQuery.ts:66]: Not using `wagmi/query`'s `useQuery` variant anymore, so we expose queryKey the same like `wagmi/query` does

File: src/AppProviders.tsx (hunk 4 above)
Comment (woodenfurniture) [reply src/AppProviders.tsx:76]: 🧠 very nice

**Review by woodenfurniture — APPROVED** [review]
tasty

**Pull request merged**